pluginLauncher/lib/source/DobbyRdkPluginManager.cpp

```diff
@@ -578,17 +666,23 @@ bool DobbyRdkPluginManager::runPlugins(const IDobbyRdkPlugin::HintFlags &hookPoi
         return false;
     }
 
-    const auto pluginsInConfig = mContainerConfig->rdk_plugins->names_of_plugins;
-    const int rdkPluginCount = mContainerConfig->rdk_plugins->plugins_count;
-
-    std::string pluginName;
+    // Determine the order of launching based on the dependencies.
+    std::vector<std::string> launchOrder;
+    if (hookPoint < IDobbyRdkPlugin::HintFlags::PostHaltFlag)
+    {
+        launchOrder = mDependencySolver.getOrderOfDependency();
+    }
+    else
+    {
+        // Reverse the order for the shutdown hooks, so that the plugins
+        // on which other plugins depend are shut down later.
+        launchOrder = mDependencySolver.getReversedOrderOfDependency();
+    }
 
     // Run all the plugins
-    for (size_t i = 0; i < rdkPluginCount; i++)
+    for (const std::string &pluginName : launchOrder)
     {
-        pluginName = pluginsInConfig[i];
-
-        const bool required = mContainerConfig->rdk_plugins->required_plugins[i];
+        const bool required = isRequired(pluginName);
 
         if (required && !isLoaded(pluginName))
```

Comment:
Do we still need all these checks here? Some of this is duplicated in the `preProcess` section now.

Reply:
Agreed, removed.